src/main/java/seedu/address/model/person/fields/subfields/NusMod.java

```diff
@@ -12,19 +14,10 @@ public class NusMod {
 
     public static final String MESSAGE_CONSTRAINTS = "Modules should be a part of NUS' NUSMods list";
     public static final String MODULELIST_FILE_PATH = "data/moduleList.json";
+    private static final Map<String, Boolean> MODULE_MAP = new HashMap<>();
     public final String name;
 
-    public NusMod(String modsString) {
-        this.name = modsString;
-    }
-
-    /**
-     * Checks whether a module is a valid NUSMod by comapring the String name to the module list data
-     *
-     * @param trimmedTag The module name to be checked
-     * @return false if the module does not exist in NUSMods
-     */
-    public static boolean isValidModName(String trimmedTag) {
+    static {
```

Comment:
Let me know if this implementation is ok. Tested it remotely and it works, I'll add in some test cases soon.